src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -250,6 +269,112 @@ private EmailWrapper generateSubmissionConfirmationEmail(
+
+        for (FeedbackSessionAttributes session : sessions) {
+            String courseId = session.getCourseId();
+            CourseAttributes course = coursesLogic.getCourse(courseId);
+            StudentAttributes student = studentsLogic.getStudentForEmail(courseId, recoveryEmailAddress);
+            StringBuilder linksFragmentValue;
+            if (linkFragmentsMap.containsKey(courseId)) {
+                linksFragmentValue = linkFragmentsMap.get(courseId);
+            } else {
+                linksFragmentValue = new StringBuilder(5000);
+            }
+
+            if (student != null) {
+                studentName = student.getName();
+                String submitUrlHtml = "";
+                String reportUrlHtml = "";
+
+                if (session.isOpened() || session.isClosed()) {
+                    String submitUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSION_SUBMISSION_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    submitUrlHtml = "[<a href=\"" + submitUrl + "\">submission link</a>]";
+                }
+
+                if (session.isPublished()) {
+                    String reportUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSION_RESULTS_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    reportUrlHtml = "[<a href=\"" + reportUrl + "\">result link</a>]";
+                }
+
+                linksFragmentValue.append(Templates.populateTemplate(
+                        EmailTemplates.FRAGMENT_SESSION_LINKS_RECOVERY_ACCESS_LINKS_BY_SESSION,
+                        "${sessionName}", session.getFeedbackSessionName(),
+                        "${submitUrl}", submitUrlHtml,
+                        "${reportUrl}", reportUrlHtml));
+
+                linkFragmentsMap.putIfAbsent(courseId, linksFragmentValue);
+            }
+        }
+
+        String recoveryUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSIONS_LINK_RECOVERY_PAGE).toAbsoluteString();
+        if (linkFragmentsMap.isEmpty()) {
+            emailBody = Templates.populateTemplate(
+                    EmailTemplates.SESSION_LINKS_RECOVERY_ACCESS_LINKS_NONE,
+                    "${teammateHomePageLink}", Config.getFrontEndAppUrl("/").toAbsoluteString(),
+                    "${userEmail}", SanitizationHelper.sanitizeForHtml(recoveryEmailAddress),
```

Comment:
below is what i got from the log msg. The userEmail is not substituted? 

<p>Hello pigshroommushpig,</p>

<p>
    You are receiving this email because a request to resend TEAMMATES links associated with the email address <strong>${userEmail}</strong> was made on <a href="http://localhost:4200/">TEAMMATES</a>.
</p>

<p>
    Here are the sessions associated with this email address within the past 180 days:
</p>

<p>Course: <strong>Sample Course 101</strong></p>
<table style="max-width:800px;min-width:500px;border-collapse: collapse; border: 1px solid black; text-align: left;">
    <tbody>
    <tr>
        <td style="border: 1px solid black; padding: 5px; vertical-align: top;"><strong>Session</strong></td>
        <td style="border: 1px solid black; padding: 5px; vertical-align: top;"><strong>Links</strong></td>
    </tr>
    <tr>
    <td style="border: 1px solid black; padding: 5px; vertical-align: top;">Sample Course 101</td>
    <td style="border: 1px solid black; padding: 5px; vertical-align: top;">[<a href="http://localhost:4200/web/sessions/submission?courseid=pigshroommushpig.gma-demo0&fsname=Sample+Course+101&key=28FFF3471BA5953D60E1A4508D662B237D1328EC3E471724A860C47F67EA07AAEE04E78C9EAB8B587283255F0B3CD8569760F4A6C2E83C7A3B0D64084929B7E9&studentemail=pigshroommushpig%40gmail.com">submission link</a>][<a href="http://localhost:4200/web/sessions/result?courseid=pigshroommushpig.gma-demo0&fsname=Sample+Course+101&key=28FFF3471BA5953D60E1A4508D662B237D1328EC3E471724A860C47F67EA07AAEE04E78C9EAB8B587283255F0B3CD8569760F4A6C2E83C7A3B0D64084929B7E9&studentemail=pigshroommushpig%40gmail.com">result link</a>]</td>
</tr>
    </tbody>
</table>
<br>

<p>
    If a session that you expected to see is missing, it may be because it is associated with a different email address. In that case, you can <a href=http://localhost:4200/web/front/help/session-links-recovery>try again</a> with a different email address or contact your course instructor(s) to check whether this email address is the one used to enroll you in TEAMMATES.
    Also, sessions that have not been opened or closed in the past 180 days are not shown. If you want the links to those sessions, please contact the TEAMMATES support team mentioned below.
</p>
<br>

<p>
    <strong>Technical help regarding TEAMMATES</strong>:
    Email TEAMMATES support team at jlinswenmogu@gmail.com.
</p>

<p>
    Regards,
    <br>TEAMMATES Team.
</p>

Reply:
That is a mistake when doing copy paste, corrected.